src/transformers/models/auto/modeling_auto.py

```diff
@@ -1655,6 +1656,17 @@ class _AutoModelWithLMHead(_BaseAutoModelClass):
 class AutoModelForCausalLM(_BaseAutoModelClass):
     _model_mapping = MODEL_FOR_CAUSAL_LM_MAPPING
 
+    @classmethod
+    def _prepare_config_for_auto_class(cls, config: PretrainedConfig) -> PretrainedConfig:
+        """
+        Additional autoclass-specific config post-loading manipulation. In this specific autoclass, if the config has
+        a nested text decoder section, uses that section instead.
+
+        Under the hood, multimodal models mapped by AutoModelForCausalLM assume the text decoder receives its own
+        config, rather than the config for the whole model.
+        """
+        return config.get_text_config(decoder=True)
```

Comment:
(this is a no-op on models without nested configs, and catches all known nested names for text decoders. See `get_text_config`: https://github.com/huggingface/transformers/blob/fc8764c9a618add64c33e83720f974750bcd0978/src/transformers/configuration_utils.py#L1120)